clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2223,6 +2225,8 @@ def err_destructor_not_member : Error<
 def err_destructor_cannot_be : Error<"destructor cannot be declared '%0'">;
 def err_invalid_qualified_destructor : Error<
   "'%0' qualifier is not allowed on a destructor">;
+def err_invalid_destructor_decl : Error<
```

Comment:
IMO, we should 'combined' the two diagnostics into `err_invalid_ctor_dtor_decl : Error<"invalid %select{constructor|destructor}0 declaration>`

Reply:
I saw that!  I'm happy once Corentin is.